### PR TITLE
prevent termination of last instance, should improve #104

### PR DESCRIPTION
**Pull request by universam1**
# Issue Type

- Bugfix Pull Request

## Summary

Prevent actively destruction of the last remaining instance. This is not intended to be an ultimate solution for #104 but should prevent harm by itself.

## Code contribution checklist

Open for any suggestion as I am new to golang.

### CODE REVIEW

**Comment by cristim** [comment]
@universam1 Thanks for your contribution to the project!

Please create some tests for the function you touched with this PR. It may not have any yet, but we're trying to add tests as we go even for existing code.

File: core/autoscaling_test.go [hunk 1]
```diff
@@ -1086,6 +1086,22 @@ func TestNeedReplaceOnDemandInstances(t *testing.T) {
...
+						Instance: &ec2.Instance{
+							State:             &ec2.InstanceState{Name: aws.String("running")},
+							Placement:         &ec2.Placement{AvailabilityZone: aws.String("eu-west-1a")},
+							InstanceLifecycle: aws.String("spot"),
+						},
+					},
+				},
+			),
+			minOnDemand:     1,
+			desiredCapacity: aws.Int64(0),
```
Comment (cristim) [inline core/autoscaling_test.go]: Shouldn't we also have a case for min=desired=1?

**Comment by universam1** [comment]
added a test to cover my piece of code. At least a little positive influence to coverage.
Please don't expect high level contribution as I started to deal with go yet since yesterday.
However, I would see this PR as a quick fix rather than a major refactor.

File: core/autoscaling_test.go (hunk 1 above)
Comment (universam1) [reply core/autoscaling_test.go]: Indeed, thanks for check

**Pull request merged**